docs/customisation.md

```diff
@@ -0,0 +1,184 @@
+
+#### **Enum**
+
+| Method | Description | Additional arguments |
+|---|---|---|
+| `item` | A method to extend enum's item. | an `item` containing the value of enum's item and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+
+### JavaScript
+
+#### **Class**
+
+| Method | Description | Additional arguments |
+|---|---|---|
+| `ctor` | A method to extend rendered constructor for a given class. | - |
+| `property` | A method to extend rendered given property. | `propertyName` as a name of a given property, `property` object as a [`CommonModel`](../src/models/CommonModel.ts) instance and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+| `setter` | A method to extend setter for a given property. | `propertyName` as a name of a given property, `property` object as a [`CommonModel`](../src/models/CommonModel.ts) instance and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+| `getter` | A method to extend getter for a given property. | `propertyName` as a name of a given property, `property` object as a [`CommonModel`](../src/models/CommonModel.ts) instance and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+
+### TypeScript
+
+#### **Class**
+
+| Method | Description | Additional arguments |
+|---|---|---|
+| `ctor` | A method to extend rendered constructor for a given class. | - |
+| `property` | A method to extend rendered given property. | `propertyName` as a name of a given property, `property` object as a [`CommonModel`](../src/models/CommonModel.ts) instance and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+| `setter` | A method to extend setter for a given property. | `propertyName` as a name of a given property, `property` object as a [`CommonModel`](../src/models/CommonModel.ts) instance and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+| `getter` | A method to extend getter for a given property. | `propertyName` as a name of a given property, `property` object as a [`CommonModel`](../src/models/CommonModel.ts) instance and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+
+#### **Interface**
+
+| Method | Description | Additional arguments |
+|---|---|---|
+| `property` | A method to extend rendered given property. | `propertyName` as a name of a given property, `property` object as a [`CommonModel`](../src/models/CommonModel.ts) instance and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+
+#### **Enum**
+
+| Method | Description | Additional arguments |
+|---|---|---|
+| `item` | A method to extend enum's item. | an `item` containing the value of enum's item and `parentModel` as a [`CommonModel`](../src/models/CommonModel.ts) instance. |
+
+#### **Type**
+
+There are no additional methods.
```

Comment:
I am just gonna put the same comment @smoya left for my PR regarding quick outdated docs. Can we add jsdoc to make these explanations?

Reply:
I also thought about that, but the problem is that types for presets are a TS interfaces. Also another options was pointing to source code of preset types, but there we have a problem, because types use generics, so it hard to read if you are not familiar with TS. In the future I have a plan for each generator to have a separate document with options/presets etc and then it will be easier to maintain.

Reply:
I tried to add jsdoc to interfaces and it doesn't work.

Reply:
Totally unfamiliar with TS jsdoc but, are we sure it is not a supported feature? I saw some issues where seems it's a thing - E.g. https://github.com/microsoft/TypeScript/issues/18358

Reply:
Yes, TS has supports for jsdoc, but interfaces is missed after transpilation to JS and even if I changed script related to render the `API.md` to `./node_modules/.bin/jsdoc2md lib/index.js -f lib/**/*.ts > API.md` then I have empty file.

Reply:
We transpile TS to JS and from JS we render docs.

Reply:
I guess that's why things like https://github.com/SoftwareBrothers/better-docs#how-it-works exist. Also, you can generate docs from TS from it afaik. Not sure what's the real effort of this, but I guess it's a topic non purely related with this PR.

Reply:
We use https://github.com/jsdoc2md/jsdoc-to-markdown so at the moment we don't want to switch :)

Reply:
@smoya maybe you can create an issue so we don't forget about better-docs package.

Reply:
> We use https://github.com/jsdoc2md/jsdoc-to-markdown so at the moment we don't want to switch :)
Fair enough 👍 

> @smoya maybe you can create an issue so we don't forget about better-docs package.

Yup, makes totally sense. Thanks for pointing that out @fmvilas

Reply:
@smoya Maybe only in https://github.com/asyncapi/generator-react-sdk. In other place we use pure JS.

Reply:
Let's just leave it there and if we prefer this one over the jsdoc-to-markdown maybe we can just replace it everywhere.